app/scripts/modules/core/projects/configure/configureProject.modal.controller.js

```diff
@@ -37,10 +37,6 @@ module.exports = angular.module('spinnaker.core.projects.configure.modal.control
       pipelines: require('./projectPipelines.modal.html'),
     };
 
-    $timeout(() => {
```

Comment:
I am guessing there's an issue with the digest cycle, where the pages aren't registered with the service when this controller initializes, hence the timeout. But your approach makes sense, too.